mk-poetry-dep.nix

```diff
@@ -214,7 +214,7 @@ pythonPackages.callPackage
                   sha256 = fileInfo.hash;
                 }
             else if isUrl then
-              builtins.fetchTarball
+              builtins.fetchurl
```

Comment:
What are the major differences between these two functions? Why is there hash mismatch when using one but not the other?

Reply:
- fetchurl fetches an url and stores whatever bytes comes back into one file in the nix store. Hash is the hash of the bytes.
- fetchTarball fetches an archive (can be zip as well, at least), and unpacks it into a folder in the nix store. Hash is the 'recursive hash' of all the unpacked files.

Poetry only records the archive's hash, not the recursive one - that's pretty nix specific in how it 's done.

Reply:
Ok, then it sounds like `fetchurl` is a closer match for poetry's behavior.